### PR TITLE
Add the Dropbox importer as a new service

### PR DESCRIPTION
**Pull request by rievo**

### CODE REVIEW

File: app/importers/dropbox.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                # Use a default, not valid token
+                self.dropbox_handler = dropbox.Dropbox(
+                    oauth2_access_token="R2D2c3p0p4dm34n4k1n3Ia"
+                )
+            else:
+                self.dropbox_handler = dropbox.Dropbox(
+                    oauth2_access_token=job.import_token
+                )
+        except (BadInputException, HttpError):
+            # self.dropbox_handler = dropbox.Dropbox(oauth2_access_token="test_token")
```
Comment (squeaky-pl) [inline app/importers/dropbox.py:73]: is this a lefrover

File: app/importers/dropbox.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+        entries = self.get_entries_from_folder(folder_url)
+
+        for entry in entries:
+            name = entry.name
+            current_level[name] = {"entry": entry, "children": {}}
+
+        for node in current_level.values():
+            entry = node.get("entry")
+            if isinstance(entry, dropbox.files.FileMetadata):
```
Comment (squeaky-pl) [inline app/importers/dropbox.py:161]: this could be removed

Comment (rievo) [reply app/importers/dropbox.py:160]: I would leave it for the next step, which would be the hooks for the files

File: app/importers/dropbox.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        url_details["path"] = "/" + search_result.group("path")
+
+    return url_details
+
+
+class DropboxImporter(BaseImporter):
+    def __init__(self, db: Session, job_id: str):
+        self.db = db
+        self.job_id = job_id
+        self.dropbox_handler: Dropbox = None
```
Comment (squeaky-pl) [inline app/importers/dropbox.py:51]:
```suggestion
        self.dropbox_handler: Optional[Dropbox] = None
```

File: app/importers/dropbox.py (hunk 1 above)
Comment (rievo) [reply app/importers/dropbox.py]: Yes, it has been removed

**Review by squeaky-pl — COMMENTED** [review]
Some things I picked up

File: app/importers/dropbox.py (hunk 2 above)
Comment (squeaky-pl) [reply app/importers/dropbox.py:160]: Ok, it's fine I guess if  you plan to extend it immediately after

File: app/tests/importers/test_dropbox.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+    importer.tree_root["entry"] = None
+
+    assert_tree_directory_recursive(
+        importer.tree_root["children"], basic_job["db_job"].path
+    )
+
+    assert job.imported_items == remote_data["total_items"]
+
+    assert job.status == JobStatus.IMPORTING_SUCCESSFULLY
```
Comment (squeaky-pl) [inline app/tests/importers/test_dropbox.py:220]:
```suggestion
    assert job.status is JobStatus.IMPORTING_SUCCESSFULLY
```

**Review by squeaky-pl — APPROVED** [review]
This looks fine from Python point of view to me. Cannot judge general direction, @drewler knows better.

**Review by drewler — APPROVED** [review]

**Pull request merged**